FakeNFT/Scenes /Catalogue/CatalogueScene/CatalogueViewController.swift

```diff
@@ -0,0 +1,138 @@
+		self.collectionListViewModel = viewModel
+	}
+	
+	private func bindViewModel() {
+		guard let viewModel = collectionListViewModel else { return }
+		viewModel.$collections.bind { [weak self] _ in
+			guard let self = self else { return }
+			self.tableView.reloadData()
+			UIBlockingProgressHUD.dismiss()
+		}
+	}
+	
+	private func setupNavBar() {
+		navigationItem.rightBarButtonItem = sortButton
+		navigationController?.navigationBar.tintColor = .black
+	}
+	
+	private func setupLayout() {
+		view.addSubview(tableView)
+		
+		NSLayoutConstraint.activate([
+			tableView.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 16),
+			tableView.trailingAnchor.constraint(equalTo: view.trailingAnchor, constant: -16),
+			tableView.topAnchor.constraint(equalTo: view.safeAreaLayoutGuide.topAnchor, constant: 20),
+			tableView.bottomAnchor.constraint(equalTo: view.safeAreaLayoutGuide.bottomAnchor)
+		])
+	}
+
+	@objc
+	private func sortCollections() {
+		let actionSheet = UIAlertController(title: "Сортировка",
+											message: nil,
+											preferredStyle: .actionSheet)
+		
+		let sortByName = UIAlertAction(title: "По названию",
+									   style: .default) { [weak self] _ in
+			guard let self = self else { return }
+			
+			UIView.animate(withDuration: 0.3) {
+				self.collectionListViewModel?.setFilterByName()
+				self.view.layoutIfNeeded()
+			}
+		}
+		
+		let sortByCount = UIAlertAction(title: "По количеству NFT",
+									   style: .default) { [weak self] _ in
+			guard let self = self else { return }
+			
+			UIView.animate(withDuration: 0.3) {
+				self.collectionListViewModel?.setFilterByCount()
+				self.view.layoutIfNeeded()
+			}
+		}
+		
+		let cancel = UIAlertAction(title: "Отмена", style: .cancel)
+		
+		actionSheet.addAction(sortByName)
+		actionSheet.addAction(sortByCount)
+		actionSheet.addAction(cancel)
+		
+		present(actionSheet, animated: true)
+	}
+}
+
+extension CollectionListViewController: UITableViewDataSource {
+	func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+		collectionListViewModel?.collections.count ?? 0
+	}
+	
+	func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+		let cell: CollectionListCell = tableView.dequeueReusableCell()
+		
+		guard let collectionItem = collectionListViewModel?.collections[indexPath.row] else { return UITableViewCell() }
+		cell.config(collectionItem: collectionItem)
+		
+		return cell
+	}
+}
+
+extension CollectionListViewController: UITableViewDelegate {
+	func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+		guard let collection = collectionListViewModel?.collections[indexPath.row] else { return }
+		
+		let collectionVC = CollectionViewController()
+		let collectionVM = CollectionViewModel()
+		collectionVC.modalPresentationStyle = .fullScreen
+		collectionVC.initialise(viewModel: collectionVM, model: collection)
```

Comment:
```suggestion
		let collectionVM = CollectionViewModel()
		let collectionVC = CollectionViewController(viewModel: collectionVM)
		collectionVC.modalPresentationStyle = .fullScreen
```

В идеале инит у контроллера должен быть такой. Я не понимаю почему еще передается модель в контроллер. Если это Model из MVVM, то ее место только во ViewModel.

Reply:
поправил